features/frameworks/unity/unity/unity.h

```diff
@@ -294,6 +295,15 @@ void tearDown(void);
 #define TEST_ASSERT_DOUBLE_IS_NOT_NAN_MESSAGE(actual, message)                                     UNITY_TEST_ASSERT_DOUBLE_IS_NOT_NAN((actual), __LINE__, (message))
 #define TEST_ASSERT_DOUBLE_IS_NOT_DETERMINATE_MESSAGE(actual, message)                             UNITY_TEST_ASSERT_DOUBLE_IS_NOT_DETERMINATE((actual), __LINE__, (message))
 
+
+/*-------------------------------------------------------
+ * Error code checking
+ *-------------------------------------------------------*/
+// Use these to check whether error code equals what we expect.
+// Only display error code (without other information)
+#define TEST_ASSERT_EQUAL_ERROR_CODE(expected, actual)                                             TEST_ASSERT_EQUAL(expected & MBED_ERROR_STATUS_CODE_MASK, actual & MBED_ERROR_STATUS_CODE_MASK)
```

Comment:
Hi David, Instead of using the error mask directly can you please use the macro MBED_GET_ERROR_CODE() macro. The reason is depending on error type the error code is encoded differently(this is due to support for Posix errors). Using the macro makes the code more portable as well to support posix errors?

Reply:
Fixed.